src/main/java/io/jenkins/plugins/opentelemetry/init/SCMEventMonitoringInitializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.jenkins.plugins.opentelemetry.init;
+
+import hudson.Extension;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import io.jenkins.plugins.opentelemetry.OpenTelemetrySdkProvider;
+import io.jenkins.plugins.opentelemetry.semconv.JenkinsSemanticMetrics;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import jenkins.YesNoMaybe;
+import jenkins.scm.api.SCMEvent;
+
+/**
+ * Capture SCM Events metrics
+ */
+@Extension(dynamicLoadable = YesNoMaybe.YES, optional = true)
```

Comment:
> (dynamicLoadable = YesNoMaybe.YES)

does the @Initializer get called if this is installed when Jenkins is already running?

Reply:
I don't think so, same issue applies to a few other metrics, you need to restart it unfortunately.

Reply:
Possibly, would rather just fix that though.

You don't happen to know how to init on plugin install dynamically loading only?

Reply:
rather hacky `Plugin.start()` is called so you can have a `start` method that checks the current init state and if Jenkins init state is complete then you know the plugin is being dynamically loaded and you can take some action.
(you can probably do the same with an `Extension` in the constructor (if the constructor is called when Jenkins is "fully up" then it has been installed post startup).

Seems rather hacky.  Would possibly be better to fix `PluginManager.dynamicLoad()` (if indeed Initializers are not called from some other place).

Reply:
Interesting. I thought that marking an Extension as dynamically loadable made it loaded during the install as well as at startup time.

Reply:
The extension will be loaded but the Initializer may not be called, so you can end up with the extension running but in an incorrect state for what it expects.

Reply:
Seems this works just fine, installing a plugin re-runs the init milestones, I verified in the debugger it gets called:

```
2022-04-19 10:40:22.223+0000 [id=114]	INFO	h.m.UpdateCenter$CompleteBatchJob#run: Completing installing of plugin batch…
2022-04-19 10:40:22.777+0000 [id=121]	INFO	jenkins.InitReactorRunner$1#onAttained: Started initialization
2022-04-19 10:40:22.780+0000 [id=126]	INFO	jenkins.InitReactorRunner$1#onAttained: Listed all plugins
2022-04-19 10:40:22.781+0000 [id=131]	INFO	jenkins.InitReactorRunner$1#onAttained: Prepared all plugins
2022-04-19 10:40:22.784+0000 [id=135]	INFO	jenkins.InitReactorRunner$1#onAttained: Started all plugins
2022-04-19 10:40:22.784+0000 [id=121]	INFO	jenkins.InitReactorRunner$1#onAttained: Augmented all extensions
2022-04-19 10:40:22.784+0000 [id=126]	INFO	jenkins.InitReactorRunner$1#onAttained: System config loaded
2022-04-19 10:40:22.784+0000 [id=131]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2022-04-19 10:40:22.785+0000 [id=123]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2022-04-19 10:40:22.785+0000 [id=128]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
2022-04-19 10:40:22.785+0000 [id=132]	INFO	jenkins.InitReactorRunner$1#onAttained: Completed initialization
2022-04-19 10:40:22.790+0000 [id=114]	INFO	h.m.UpdateCenter$CompleteBatchJob#run: Completed installation of 1 plugins in 3.2 sec
```